### PR TITLE
Remove static config and Kind trait conversion

### PR DESCRIPTION
**Pull request by npmenard**
This covers RSDK-5875 and RSDK-2321, plus a small refactor of the config logic

### CODE REVIEW

File: src/common/config.rs [hunk 1]
```diff
@@ -399,35 +294,26 @@ impl TryFrom<&ComponentConfig> for DynamicComponentConfig {
 }
 
 #[derive(Debug)]
-pub enum ConfigType {
-    Static(&'static StaticComponentConfig),
-    Dynamic(DynamicComponentConfig),
+pub enum ConfigType<'a> {
```
Comment (gvaradarajan) [inline src/common/config.rs:297]: should we just remove the enum, use `DynamicComponentConfig` directly and rename to `ComponentConfig`?

File: examples/esp32/partitions.csv [hunk 2]
```diff
@@ -1,5 +1,5 @@
 # Name,   Type, SubType, Offset,  Size, Flags
 # Note: if you have increased the bootloader size, make sure to update the offsets to avoid overlap
-nvs,      data, nvs, 0x9000    ,        0x8000,
-phy_init, data, phy, 0x11000    ,        0x1000,
-factory,  app,  factory, 0x30000,       0x3D0000,
+nvs,      data, nvs, 0x9000    ,        0x6000,
+phy_init, data, phy, 0xf000    ,        0x1000,
```
Comment (gvaradarajan) [inline examples/esp32/partitions.csv:4]: [q] how often should we check that the updated compiled binary fits in the factory partition?

**Review by gvaradarajan — CHANGES_REQUESTED** [review]
LGTM, I'm glad we're finally doing this. Just one adjustment I think worth doing

File: examples/esp32/partitions.csv (hunk 2 above)
Comment (npmenard) [reply examples/esp32/partitions.csv:4]: We should be mindful we don't want the binary to bloat even in Debug mode

**Review by acmorrow — APPROVED** [review]
LGTM. I went through it pretty quickly because it is mostly a feature removal and associated simplifications.

File: src/common/config.rs (hunk 1 above)
Comment (gvaradarajan) [reply src/common/config.rs:297]: understood, I'm comfortable with that

**Review by gvaradarajan — APPROVED** [review]
LGTM

**Pull request merged**